### PR TITLE
Enable nullable: System.Management.Automation.Language.ICustomAstVisitor

### PR DESCRIPTION
**Pull request by powercode**
Tracking issue: #12631.

### CODE REVIEW

File: src/System.Management.Automation/engine/parser/AstVisitor.cs [hunk 1]
```diff
@@ -8,198 +8,203 @@
...
 
         /// <summary/>
-        object VisitBlockStatement(BlockStatementAst blockStatementAst) => DefaultVisit(blockStatementAst);
+        object? VisitBlockStatement(BlockStatementAst blockStatementAst) => DefaultVisit(blockStatementAst);
 
         #endregion Expressions
     }
 
+#nullable restore
+
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/parser/AstVisitor.cs:207]:
```suggestion
#nullable restore

```

File: src/System.Management.Automation/engine/parser/AstVisitor.cs [hunk 2]
```diff
@@ -8,198 +8,203 @@
 
 namespace System.Management.Automation.Language
 {
+
+#nullable enable
+
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/parser/AstVisitor.cs:13]:
```suggestion
```

File: src/System.Management.Automation/engine/parser/AstVisitor.cs [hunk 3]
```diff
@@ -8,198 +8,203 @@
...
     /// <summary>
     /// </summary>
     public interface ICustomAstVisitor
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/parser/AstVisitor.cs:16]:
```suggestion
#nullable enable
    public interface ICustomAstVisitor
```

**Review by iSazonov — COMMENTED** [review]
Looks good. Only please move the directive.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@powercode Please add the standard XML comment like "This never returns null." to appropriate methods.

**Review by rjmholt — CHANGES_REQUESTED** [review]
> @powercode Please add the standard XML comment like "This never returns null." to appropriate methods.

Blocked on this

**Review by rjmholt — APPROVED** [review]

**Pull request merged**